### ISSUE TITLE
Why set TargetExt to .dll even if the OutputType is Exe on .NETCoreApp? 

### ISSUE THREAD

**Issue by haiyuzhu** [dotnet/sdk#15392]
I found `TargetExt` was set to `.dll` on .NETCoreApp projects in [Microsoft.NET.Sdk.BeforeCommon.targets](https://github.com/dotnet/sdk/blame/master/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.BeforeCommon.targets) even if the `OutputType` is `Exe`. Could anybody help me understand the story here？ Thanks
```xml
    <!-- Force .dll extension for .NETCoreApp and .NETStandard projects even if output type is exe. -->
    <TargetExt Condition="'$(TargetExt)' == ''">.dll</TargetExt>
```

**Comment by AraHaan** [comment]
NAB, it is intended as it outputs an stub (dotnet.exe customized with resources added to the projected itself) that becomes the actual exe that loads the dll on that.

**Comment by dasMulli** [comment]
A similar discussion happend in the 2.0 timeframe b/c `Exe` as output type has nothing to do with the actual file extension and just means "i intend to run this thing and it therefore should have an entry point (Main method) and anything needed to run it". For portable apps this means a `.dll` file, for framework-dependent executables or self-contained apps no extensions (linux, macOS) or a `.exe`.
The bottom line is that an awful lot of build tooling out there looks for the known values of `OutputType`.

There now is also `$(HasRuntimeOutput)` (true/false) to look for which is probably more useful as it also covers things like `WinExe`

**Comment by haiyuzhu** [comment]
Thank you very much for the explanation. Close the issue.
